plugins/lenovo-thinklmi/fu-plugin-lenovo-thinklmi.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2021 Mario Limonciello <mario.limonciello@amd.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	g_autofree gchar *sysfsfwdir = NULL;
+	g_autofree gchar *thinklmidir = NULL;
+
+	/* already exists */
```

Comment:
do we need a kernel_version check here? No is a valid answer.

Reply:
No, older kernel and not a Lenovo system just won't run the plugin, should abort startup.